src/server/tweetFinder.js

```diff
@@ -318,3 +306,26 @@ export const getTweetFromDb = (res, id) => (
       )
     )
 );
+
+/**
+ * Used in ./twitterSearch.js:169.
+ * Tries to get the missing user tweeter for the original tweet from DB
+ * If not in DB then make a call to twitter to retrieve the whole tweet that contains the user tweeter
+ * @param String representing the original tweet's id
+ * @return Tweeter An Immutable Object representing the tweet's user
+ */
+export const getOriginalTweetUserFromTweet = (tweetId) =>
+  newPromiseChain()
+    .then(() => (
+       db.query(
+         makeTweetQuerySelectingFrom('SELECT FROM tweet WHERE id =:id'), { params: { id: tweetId, limit: 1 }, }
```

Comment:
For the record, "limi: 1" will only work if you add LIMIT :limit after the WHERE